### PR TITLE
Upgrade jQuery to v3.5.1, refs #13453

### PR DESCRIPTION
**Pull request by jraddaoui**
- Upgrade jQuery and jQuery UI to their latest versions.
- Fix incompatibilities and some deprecations across all JS code.
- Clean-up and remove unused front-end assets from:
   - sfDrupalPlugin.
   - sfFormExtraPlugin.
   - YUI.
   - jQuery UI.
- Minor tweak in Bootstrap 2.3.2 dropdown code.
- Clean assets loaded dinamically and via config files.
- Remove unused JS files (actions, contextMenu).
- Remove sfFormExtraPlugin.
- Modify unavailable page style to match the error page.
- Use jQuery UI datepicker in accession/deaccession forms.

### CODE REVIEW

File: vendor/bootstrap/js/bootstrap.js [hunk 1]
```diff
@@ -756,7 +756,8 @@
       selector = selector && /#/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') //strip for ie7
     }
 
-    $parent = selector && $(selector)
+    // Removed for jQuery 3.5.1
```
Comment (sbreker) [inline vendor/bootstrap/js/bootstrap.js:759]: Nitpick: commented code

File: plugins/sfDrupalPlugin/vendor/drupal/misc/collapse.js [hunk 2]
```diff
@@ -65,10 +65,10 @@ Drupal.behaviors.collapse = {
   attach: function (context, settings) {
     $(location.hash, context).removeClass('collapsed');
 
-    $('fieldset.collapsible > legend', context).once('collapse', function () {
+    $('fieldset.collapsible > legend', context).each(function () {
```
Comment (sbreker) [inline plugins/sfDrupalPlugin/vendor/drupal/misc/collapse.js:68]: Great work on these jquery updates!

File: plugins/sfDrupalPlugin/vendor/drupal/misc/collapse.js [hunk 3]
```diff
@@ -65,10 +65,10 @@ Drupal.behaviors.collapse = {
...
       var fieldset = $(this.parentNode);
       // Expand if there are errors inside.
-      if ($('input.error, textarea.error, select.error', fieldset).size() > 0) {
+      if ($('input.error, textarea.error, select.error', fieldset).length > 0) {
```
Comment (sbreker) [inline plugins/sfDrupalPlugin/vendor/drupal/misc/collapse.js:71]: 👍

**Review by sbreker — APPROVED** [review]
Great work @jraddaoui! Was every module affected by the changes? Wondering if it would be possible to focus testing on specific modules?

**Comment by jraddaoui** [comment]
Thanks @sbreker! Yes, pretty much every module is touched by this changes, I'm adding notes in the Redmine ticket for testing.

File: vendor/bootstrap/js/bootstrap.js (hunk 1 above)
Comment (jraddaoui) [reply vendor/bootstrap/js/bootstrap.js:759]: This is intentional and the only change needed (for now) to make the Bootstrap components work with jQuery 3.

**Pull request merged**